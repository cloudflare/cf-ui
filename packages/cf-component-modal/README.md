# cf-component-modal

> Cloudflare Modal Component

## Installation

```sh
$ npm install --save cf-component-modal react-gateway
```

## Usage

```js
const React = require('react');
const {
  Modal,
  ModalHeader,
  ModalTitle,
  ModalClose,
  ModalBody,
  ModalFooter,
  ModalActions
} = require('cf-component-modal');
const {
  GatewayDest,
  GatewayProvider
} = require('react-gateway');

class MyComponentWithModal extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isModalOpen: false
    };
  }

  handleRequestOpen() {
    this.setState({ isModalOpen: true });
  }

  handleRequestClose() {
    this.setState({ isModalOpen: false });
  }

  render() {
    return (
      <div>
        <button onClick={this.handleRequestOpen.bind(this)}>Open Modal</button>
        <Modal
          isOpen={this.state.isModalOpen}
          onRequestClose={this.handleRequestClose.bind(this)}>
          <ModalHeader>
            <ModalTitle>Hello from Modal</ModalTitle>
            <ModalClose onClick={this.handleRequestClose.bind(this)}/>
          </ModalHeader>
          <ModalBody>
            <p>Look at this awesome modal!</p>
          </ModalBody>
          <ModalFooter>
            <ModalActions>
              <button onClick={this.handleRequestClose.bind(this)}>Close Modal</button>
            </ModalActions>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

class Application extends React.Component {
  render() {
    return (
      <GatewayProvider>
        <div className="application">
          <div className="application__content">
            <h1>Cloudflare Modal Component Example</h1>
            <MyComponentWithModal/>
          </div>
          <GatewayDest name="modal"/>
        </div>
      </GatewayProvider>
    );
  }
}
```
