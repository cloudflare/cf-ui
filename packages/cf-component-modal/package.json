{
  "name": "cf-component-modal",
  "description": "Cloudflare Modal Component",
  "version": "6.8.0",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "author": "James Kyle <jkyle@cloudflare.com>",
  "license": "BSD-3-Clause",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "dependencies": {
    "cf-component-icon": "^2.4.1",
    "cf-style-container": "^1.2.0",
    "prop-types": "^15.5.8",
    "react-gateway": "^2.2.0",
    "react-modal2": "^3.1.0",
    "react-transition-group": "^1.1.1"
  },
  "peerDependencies": {
    "react": "^0.14.9 || ^15.0.0-0 || ^16.0.0-0"
  }
}
