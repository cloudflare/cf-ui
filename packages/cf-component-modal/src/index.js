import Modal from './Modal';
import ModalActions from './ModalActions';
import ModalBody from './ModalBody';
import ModalClose from './ModalClose';
import ModalFooter from './ModalFooter';
import ModalHeader from './ModalHeader';
import ModalTitle from './ModalTitle';

export {
  Modal,
  ModalActions,
  ModalBody,
  ModalClose,
  ModalFooter,
  ModalHeader,
  ModalTitle
};
