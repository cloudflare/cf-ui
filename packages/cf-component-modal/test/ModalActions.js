import React from 'react';
import renderer from 'react-test-renderer';
import { ModalActions } from 'cf-component-modal';

test('should render', () => {
  const component = renderer.create(<ModalActions>ModalActions</ModalActions>);
  expect(component.toJSON()).toMatchSnapshot();
});
