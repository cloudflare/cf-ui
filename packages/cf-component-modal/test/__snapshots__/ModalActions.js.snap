// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="cf-modal__actions"
>
  ModalActions
</div>
`;
